cumulusci/tasks/github/tests/test_merge.py

```diff
@@ -335,162 +347,462 @@ def test_main_parent_with_child_pr(self):
+
+        prefix = "neptune/"
+        source_branch = "main"
+        child_branches = [
+            f"{prefix}230",
+            f"{prefix}work-a",
+            f"{prefix}work-b",
+        ]
+        other_branches = [
+            "venus/work-a",
+            f"{prefix}work-a__child_a",
+            f"{prefix}work-a__child_a__grandchild",
+            f"{prefix}work-b__child_b",
+            f"{prefix}orphan__with_child",
+            f"{prefix}230__cool_feature",
+            f"{prefix}230__cool_feature__child",
+        ]
+        self._setup_mocks([source_branch] + child_branches + other_branches)
 
         with LogCapture() as log:
-            task = self._create_task()
+            task = self._create_task(
+                task_config={
+                    "options": {"source_branch": source_branch, "branch_prefix": prefix}
+                }
+            )
             task()
 
-            log_lines = self._get_log_lines(log)
-
-            expected = [
+            expected_log = [
                 ("INFO", "Beginning task: MergeBranch"),
                 ("INFO", ""),
-                ("DEBUG", "Skipping branch main: is source branch"),
+                ("DEBUG", f"Skipping branch {source_branch}: is source branch"),
+                (
+                    "DEBUG",
+                    f"Skipping branch {other_branches[0]}: does not match prefix '{prefix}'",
+                ),
+                (
+                    "DEBUG",
+                    f"Skipping branch {other_branches[1]}: is not a direct descendent of {source_branch}",
+                ),
+                (
+                    "DEBUG",
+                    f"Skipping branch {other_branches[2]}: is not a direct descendent of {source_branch}",
+                ),
+                (
+                    "DEBUG",
+                    f"Skipping branch {other_branches[3]}: is not a direct descendent of {source_branch}",
+                ),
+                (
+                    "DEBUG",
+                    f"Skipping branch {other_branches[4]}: is not a direct descendent of {source_branch}",
+                ),
+                (
+                    "DEBUG",
+                    f"Skipping branch {other_branches[5]}: is not a direct descendent of {source_branch}",
+                ),
+                (
+                    "DEBUG",
+                    f"Skipping branch {other_branches[6]}: is not a direct descendent of {source_branch}",
+                ),
+                (
+                    "INFO",
+                    f"Merged 1 commits into branch: {child_branches[0]}",
+                ),
                 (
                     "INFO",
-                    "Merged 1 commits into parent branch {}".format(parent_branch_name),
+                    f"Merged 1 commits into branch: {child_branches[1]}",
+                ),
+                (
+                    "INFO",
+                    f"Merged 1 commits into branch: {child_branches[2]}",
```

Comment:
In general in tests I would aim for abstracting things out into functions and variables less. It would be easier for me to read this and verify that the output makes sense if this had the expected branch names inline instead of referring to a variable I have to jump around to look at.

Reply:
Also perhaps we could simplify all these tests whose purpose is to verify that the correct branches are selected, by checking the result of task._get_branches_to_merge() rather than running the entire task and looking at what was logged.